src/Bicep.Core/Emit/ExpressionEmitter.cs

```diff
@@ -182,6 +182,47 @@ public void EmitObjectProperties(ObjectSyntax objectSyntax, ISet<string>? proper
             }
         }
 
+        public void EmitModuleParameterValue(SyntaxBase syntax)
+        {
+            if (syntax is InstanceFunctionCallSyntax instanceFunctionCall)
+            {
+                var objectSymbol = context.SemanticModel.GetSymbolInfo(instanceFunctionCall.BaseExpression);
+                if (objectSymbol is ResourceSymbol resource && resource.Type is ResourceType resourceType &&
+                            resourceType.Body is NamedObjectType objectType &&
+                            string.Equals(objectType.Name, "Microsoft.KeyVault/vaults", StringComparison.OrdinalIgnoreCase))
```

Comment:
Note to change this to `StringComparer.OrdinalIgnoreCase.Equals(` and extract `Microsoft.KeyVaylt/vaults` to a constant in your final PR

Reply:
extracting to constant - yes, but is there a difference between `string.equals(v1,v2, StringComparer.OrdinalIngnoreCase)` and `StringComparer.OrdinalIgnoreCase.Equals`?

Reply:
@TarunSunkaraneni  I simplified this a bit, Now i just check for the function name, not for the type on which it's being called.